sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuShuffleCoalesceExec.scala

```diff
@@ -194,13 +205,80 @@ class JCudfTableOperator extends SerializedTableOperator[SerializedTableColumn]
   }
 }
 
+case class KudoHostMergeResultWrapper(inner: KudoHostMergeResult,
+    dataSize: Long) extends CoalescedHostResult {
+
+  /** Convert itself to a GPU batch */
+  override def toGpuBatch(dataTypes: Array[DataType]): ColumnarBatch = {
+    RmmRapidsRetryIterator.withRetryNoSplit {
+      closeOnExcept(inner.toTable) { cudfTable =>
+        GpuColumnVector.from(cudfTable, dataTypes)
+      }
+    }
+  }
+
+  /** Get the data size */
+  override def getDataSize: Long = dataSize
+
+  override def close(): Unit = inner.close()
+}
+
+case class RowCountOnlyMergeResult(rowCount: Int) extends CoalescedHostResult {
+  override def toGpuBatch(dataTypes: Array[DataType]): ColumnarBatch = {
+    new ColumnarBatch(Array.empty, rowCount)
+  }
+
+  override def getDataSize: Long = 0
+
+  override def close(): Unit = {}
+}
+
+class KudoTableOperator(
+    kudo: Option[KudoSerializer] ,
+    kudoMergeHeaderTime: GpuMetric,
+    kudoMergeBufferTime: GpuMetric,
+) extends
+  SerializedTableOperator[KudoSerializedTableColumn] {
+  require(kudo != null, "kudo serializer should not be null")
+
+  override def getDataLen(column: KudoSerializedTableColumn): Long = column
+    .kudoTable
```

Comment:
The code format is a little of strange, and does not match our current style.
how about putting them in a single line ?